src/main.c

```diff
@@ -2230,252 +489,181 @@ UX_FLOW(ux_sign_flow,
-    UX_DISPLAY(ui_idle_blue, NULL);
-#elif defined(TARGET_NANOS)
-    UX_MENU_DISPLAY(0, menu_main, NULL);
-#elif defined(TARGET_NANOX) || defined(TARGET_NANOS2)
+
     // reserve a display stack slot if none yet
     if(G_ux.stack_count == 0) {
         ux_stack_push();
     }
     ux_flow_init(0, ux_idle_flow, NULL);
-#endif // #if TARGET_ID
 }
 
-#if defined(TARGET_BLUE)
-unsigned int io_seproxyhal_touch_settings(const bagl_element_t *e) {
-    UX_DISPLAY(ui_settings_blue, ui_settings_blue_prepro);
-    return 0; // do not redraw button, screen has switched
+#else // HAVE_BAGL
+
+void ui_idle(void) 
+{
+    skipDataWarning = false;
+    skipClausesWarning = false;
+    ui_menu_main();
+}
+#endif
+
+
+//////////////////////////////// Crypto functions //////////////////////////////////////
+
+int crypto_derive_private_key(cx_ecfp_private_key_t *private_key,
+                              uint8_t *chain_code,
+                              const uint32_t *bip32_path,
+                              uint8_t bip32_path_len) {
+    uint8_t raw_private_key[32] = {0};
+    int error = 0;
+
+    BEGIN_TRY {
+        TRY {
+            // derive the seed with bip32_path
+            os_perso_derive_node_bip32(CX_CURVE_256K1,
+                                       bip32_path,
+                                       bip32_path_len,
+                                       raw_private_key,
+                                       chain_code);
+            // new private_key from raw
+            cx_ecfp_init_private_key(CX_CURVE_256K1,
+                                     raw_private_key,
+                                     sizeof(raw_private_key),
+                                     private_key);
+        }
+        CATCH_OTHER(e) {
+            error = e;
+        }
+        FINALLY {
+            explicit_bzero(&raw_private_key, sizeof(raw_private_key));
+        }
+    }
+    END_TRY;
+
+    return error;
+}
+
+void crypto_init_public_key(cx_ecfp_private_key_t *private_key,
+                            cx_ecfp_public_key_t *public_key,
+                            uint8_t raw_public_key[static 64]) {
+    // generate corresponding public key
+    cx_ecfp_generate_pair(CX_CURVE_256K1, public_key, private_key, 1);
+
+    memmove(raw_public_key, public_key->W + 1, 64);
+}
+
+int crypto_sign_message() 
+{
+    cx_ecfp_private_key_t private_key = {0};
+    uint32_t info = 0;
+   
+    // derive private key according to BIP32 path
+    int error = crypto_derive_private_key(&private_key,
+                                          NULL,
+                                          tmpCtx.transactionContext.bip32Path,
+                                          tmpCtx.transactionContext.pathLength);
+                                          
+    if (error != 0) {
+        return error;
+    }
+
+    BEGIN_TRY {
+        TRY {
+            cx_ecdsa_sign(&private_key,
+                          CX_RND_RFC6979 | CX_LAST,
+                          CX_SHA256,
```

Comment:
Should be `CX_BLAKE2B`

Reply:
not sure. It is not in the scope of Stax porting. As discussed, you could open a subsequently PR to address this point if it is required.